### DISCUSSION TITLE
Installing pai in a python venv (virtual environment)

### DISCUSSION THREAD

**Discussion by gergelyszabo94** [ParadoxAlarmInterface/pai#490] [Q&A] [answered]
## Alarm system
Panel: MG5050, access mode IP150+

## Environment
Linux, RPI, Homebrdige image...
Python version: 3.11.2

### Git branch
 master

## Question
I am running Homebridge on my RPI5, I installed the homebridge image directly, no docker. I managed to install mosquitto successfully, but I am having issues with this step: sudo -H pip3 install paradox-alarm-interface . I get this: "error: externally-managed-environment

× This environment is externally managed

To install Python packages system-wide, try apt install
  python3-xyz, where xyz is the package you are trying to
  install.

  If you wish to install a non-Debian-packaged Python package,
  create a virtual environment using python3 -m venv path/to/venv.
  Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
  sure you have python3-full installed.

  For more information visit http://rptl.io/venv"

I created a virtual envirtonment and installed pai there, but now I am stuck, unsure how to proceed since pai can't find the configuration file. Am I on the right path here, should I try making it work within a venv or am I totally off course?

**Comment by yozik04** [comment, marked as answer]
You do all correctly. You can pass in configuration file location. Run PAI with --help it will tell you possible arguments. I think you will need to pass in -c option with configuration file location.

**Comment by gergelyszabo94** [comment]
@yozik04 You are right, I should be able to set the configuration file location using the -c option, butI can't find my configuration file.
I did:
```
python3 -m venv --system-site-packages paivenv
source paivenv/bin/activate
pip3 install paradox-alarm-interface
```
inside the homebridge directory. So after activating the venv it looks like this:
```(paivenv) pi@homebridge:/var/lib/homebridge/paivenv $```
I can't find the config.

**Comment by yozik04** [comment]
Copy from default one.
and follow the doc https://github.com/ParadoxAlarmInterface/pai/wiki/Configuration

**Comment by gergelyszabo94** [comment]
Thanks, I managed to connect, I was silly and looking for the config on disk and not in the repo. If you don't mind, leave this discussion open for a few days while I make everything work, I am still unsure how it will all work with homebridge.

**Comment by gergelyszabo94** [comment]
I decided to do it in docker instead and it seems to be working, thanks for the help.
